ssf/samples.go

```diff
@@ -102,8 +115,19 @@ func create(base *SSFSample, opts []SampleOption) *SSFSample {
 func RandomlySample(rate float32, samples ...*SSFSample) []*SSFSample {
 	res := make([]*SSFSample, 0, len(samples))
 
+	randFloat := defaultRandFloat32
+
+	r, ok := rngPool.Get().(*rand.Rand)
+	if ok {
+		randFloat = func() float32 {
```

Comment:
For risk of showing that I'm not sure I know what I'm talking about, could you do
``` go
randFloat = r.Float32
```
here instead? I guess the closure will get stack-allocated (or doesn't even allocate) anyway, but seems like a lot of syntax in either case.

Reply:
https://play.golang.org/p/GWzPCIrne6E indicates you could...

Reply:
It doesn't allocate, but yes, we can! There's a small performance improvement from not creating the closure.